src/types.rs

```diff
@@ -345,12 +357,16 @@ where
 
     let item_vec: Vec<_> = items.collect();
 
-    let tactic = definitive_tactic(
-        &*item_vec,
-        ListTactic::HorizontalVertical,
-        Separator::Comma,
-        budget,
-    );
+    let tactic = if output.contains('\n') {
```

Comment:
If I understand correctly this bit, can you add a comment explaining that a function's argument are forced to be formatted vertically in case the the return type spans multiple lines ?